sridha2412/assignment2/q1.py

```diff
@@ -8,24 +8,26 @@ def __init__(self, data: int):
         self.left = None
         self.right = None
 
-def find_ancestors(root: Node, key: int) -> bool:
+def find_ancestors(root: Node, key: int, ancestors: list) -> bool:
     """ if key is present, return list of ancestors else return false """
     # base cases
     if root is None:
         return False
```

Comment:
In Python, key is not present is spelled KeyError. Instead of returning values that means "something went wrong", the convention is to raise an exception. So instead of:

return False

you write:

raise KeyError('value %s not found in the tree' % key)